worker/include/RTC/ActiveSpeakerObserver.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef MS_RTC_ACTIVE_SPEAKER_OBSERVER_HPP
+#define MS_RTC_ACTIVE_SPEAKER_OBSERVER_HPP
+
+#include "RTC/RtpObserver.hpp"
+#include "handles/Timer.hpp"
+#include <nlohmann/json.hpp>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+// Implementation of Dominant Speaker Identification for Multipoint
+// Videoconferencing by Ilana Volfin and Israel Cohen. This
+// implementation uses the RTP Audio Level extension from RFC-6464
+// for the input signal.
+// This has been ported from DominantSpeakerIdentification.java in Jitsi.
+// https://github.com/jitsi/jitsi-utils/blob/master/src/main/java/org/jitsi/utils/dsi/DominantSpeakerIdentification.java
+namespace RTC
+{
+	class ActiveSpeakerObserver : public RTC::RtpObserver, public Timer::Listener
+	{
+	private:
+		class Speaker
+		{
+		public:
```

Comment:
Please let reorder this as we do in other C++ classes:

1. Public static methods first.
2. Then private static methods.
3. Then public static members.
4. Then private static members.
5. Public class methods first.
6. Then private class methods.
7. Then public class members.
8. Then private class members.

This is:

```c++
public:
	Speaker();
	void EvalActivityScores();
	double GetActivityScore(int32_t interval);
	void LevelChanged(uint32_t level, uint64_t time);
	void LevelTimedOut();

private:
	bool ComputeImmediates();
	bool ComputeLongs();
	bool ComputeMediums();
	void EvalImmediateActivityScore();
	void EvalMediumActivityScore();
	void EvalLongActivityScore();
	void UpdateMinLevel(int8_t level);

public:
	bool paused{ false };
	double immediateActivityScore;
	double mediumActivityScore;
	double longActivityScore;
	uint64_t lastLevelChangeTime{ 0 };

private:
	uint8_t minLevel;
	uint8_t nextMinLevel;
	uint32_t nextMinLevelWindowLen{ 0 };
	std::vector<uint8_t> immediates;
	std::vector<uint8_t> mediums;
	std::vector<uint8_t> longs;
	std::vector<uint8_t> levels;
```